sympy/solvers/solveset.py

```diff
@@ -1718,10 +1723,30 @@ def linear_eq_to_matrix(equations, *symbols):
         # Extract coeff of symbols
         coeff_list = []
         for symbol in symbols:
-            coeff_list.append(f.coeff(symbol))
+            coeff_list.append(f.diff(symbol))
 
         # append constant term (term free from symbols)
         coeff_list.append(-f.as_coeff_add(*symbols)[0])
```

Comment:
If you follow the suggestions above, you will just append `-f` because all the terms involving any symbol of interest have been removed.